### PR TITLE
Remove unused flags from old linking approach

### PR DESCRIPTION
**Pull request by mithydolphin**
With the transition to the new linking principle now complete, this commit removes
flags that were needed for the previous approach. These flags are no longer relevant
or used in the current implementation.

Change-Id: I6adb35497df6a7845225534a9ac4fe18980007c7

### Describe your changes

Please include a summary of the change, a relevant motivation and context.

### Issue ticket number and link

- Fixes #(issue)

### Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have verified that the code builds perfectly fine on my local system
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have verified that my code follows the style already available in the repository
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: lib/src/signed_video_h26x_auth.c [hunk 1]
```diff
@@ -1025,18 +1023,6 @@ maybe_validate_gop(signed_video_t *self, h26x_nalu_t *nalu)
       // validation in the middle of a stream. Now it is time to reset it.
       validation_flags->is_first_validation = !validation_flags->signing_present;
 
-      if (validation_flags->reset_first_validation) {
-        validation_flags->is_first_validation = true;
```
Comment (bjornvolcker) [inline lib/src/signed_video_h26x_auth.c:1029]: Should you really remove the reset? This is used to handle the case where there are more than one SEI in the beginning of a stream that cannot be used. The scrapped test for example.

**Review by bjornvolcker — CHANGES_REQUESTED** [review]

File: lib/src/signed_video_h26x_auth.c (hunk 1 above)
Comment (mithydolphin) [reply lib/src/signed_video_h26x_auth.c:1029]: But as I understand its for the first validation and  resetting  the first validation is done without a flag in h26x_nalu_list_remove_missing_items function.

Comment (bjornvolcker) [reply lib/src/signed_video_h26x_auth.c:1029]: Hmm, I'm not 100% convinced. Here is an example
If `IPIPPIPPPIPPPPIP`
is signed into
`IPIPPISPSPPISPPPPISP`
and remove the first 2 GOPs to
`IsPsPPISPPPPISP`
then the first 2 `s` do not belong to this recording. In this case the first 2 validations need to be discarded.
Make a test of this

**Review by bjornvolcker — CHANGES_REQUESTED** [review]

File: tests/check/check_signed_video_auth.c [hunk 2]
```diff
@@ -1053,6 +1053,67 @@ START_TEST(add_one_sei_nalu_after_signing)
...
+      SV_AUTH_RESULT_NOT_OK, false, 19, 16, 3, SV_PUBKEY_VALIDATION_NOT_FEASIBLE, true, 0, 0};
+  // ISPPSPISPPPPISPPISP
+  //
+  // IS                PU                      -> (signature)
+  // ISPPS             NUNN.                   -> (invalid)
+  //     PIS           MMMNP.                  -> (invalid)
+  //      ISPPPPIS              ......P.       -> (valid)
+  //            ISPPIS                 ....P.  -> (valid)
+  struct validation_stats expected = {.valid_gops = 2,
+      .invalid_gops = 2,
```
Comment (bjornvolcker) [inline tests/check/check_signed_video_auth.c:1095]: Nothing is wrong with this stream. I would expect all of them to be valid.

**Review by bjornvolcker — CHANGES_REQUESTED** [review]

**Review by bjornvolcker — APPROVED** [review]

**Pull request merged**